web/thesauruses/kotlin/1.5/classes.json

```diff
@@ -0,0 +1,112 @@
+      },
+      "class_with_generic_type": {
+        "code": "class ClassName<T> {\n var property: T\n}",
+        "comment": "Kotlin generic classes are defined with a type parameter in angle brackets.",
+        "name": "Class with a generic type"
+      },
+      "private_variables": {
+        "code": "private var variableName: Type = value",
+        "name": "Defining private variables"
+      },
+      "protected_variables": {
+        "code": "protected var variableName: Type = value",
+        "name": "Defining protected variables"
+      },
+      "public_variables": {
+        "code": "var variableName: Type = value",
+        "comment": "Variables are public by default in Kotlin.",
+        "name": "Defining public variables"
+      },
+      "static_variables": {
+        "code": "class ClassName {\n companion object {\n var staticVar: Type = value\n}\n}",
+        "name": "Defining static variables"
+      },
+      "private_functions": {
+        "code": "private fun methodName() {\n // Method body \n}",
+        "name": "Defining private functions"
+      },
+      "protected_functions": {
+        "code": "protected fun methodName() {\n // Method body \n}",
+        "name": "Defining protected functions"
+      },
+      "public_functions": {
+        "code": "fun methodName() {\n // Method body \n}",
+        "name": "Defining public functions"
+      },
+      "static_functions": {
+        "code": "class ClassName {\n companion object {\n fun methodName() {\n // Method body \n}\n}\n}",
+        "name": "Defining static functions"
+      },
+      "extends_class": {
+        "code": "class ChildClassName : ParentClassName() {\n // Class body \n}",
+        "name": "Class that inherits/extends another class"
+      },
+      "extending_interface": {
+        "code": "class ClassName : InterfaceName {\n // Class body \n}",
+        "name": "Class/Interface that inherits/extends another class/interface"
+      },
+      "calling_superclass_functions": {
+        "code": "super.methodName()",
+        "name": "Calling a superclass function"
+      },
+      "overriding_superclass_functions": {
+        "code": "override fun methodName() {\n // Overridden method body \n}",
+        "name": "Overriding a superclass function"
+      },
+      "instantiating_object": {
+        "code": "val obj = ClassName()",
+        "name": "Instantiating a new object"
+      },
+      "instantiating_polymorphic_object": {
+        "code": "val obj: InterfaceName = ClassName()",
+        "comment": "An object can be instantiated as a polymorphic type.",
+        "name": "Instantiating a polymorphic object"
+      },
+      "implement_constructor": {
+        "code": "class ClassName(val property: Type) {\n init {\n // Constructor body \n}\n}",
+        "name": "Implementing a class constructor"
+      },
+      "implement_deconstructor": {
+        "not-implemented": true,
+        "comment": "Kotlin does not have destructors like C++.",
```

Comment:
[NB] You don't have to have this comment if it's not implemented since it doesn't provide anything extra

Reply:
I have removed the unnecessary comment from the `implement_deconstructor` concept and fixed the formatting for the code snippets to ensure proper indentation.